### PR TITLE
feat(dh): group delegations in overview

### PR DESCRIPTION
**Pull request by dzhavat**
## References

- N/A

### CODE REVIEW

File: libs/dh/market-participant/actors/feature-actors/src/lib/drawer/dh-actor-drawer.component.html [hunk 1]
```diff
@@ -96,7 +96,11 @@ <h2 class="actor-heading">{{ actor?.glnOrEicNumber }} {{ actor?.name }}</h2>
               [label]="t('tabs.delegation.tabLabel')"
               *dhPermissionRequired="['delegation:view']"
             >
-              <dh-delegation-tab />
+              @defer (on viewport) {
```
Comment (mimse) [inline libs/dh/market-participant/actors/feature-actors/src/lib/drawer/dh-actor-drawer.component.html:99]: Praise: nice :-)

File: libs/dh/market-participant/actors/feature-delegation/src/lib/dh-delegation-tab.component.ts [hunk 2]
```diff
@@ -37,21 +44,35 @@ import { DhDelegationCreateModalComponent } from './dh-delegation-create-modal.c
...
-          >
-            {{ t('emptyStateCreateBtn') }}
-          </watt-button>
-        </watt-empty-state>
-      </vater-stack>
+      @if (isLoading()) {
+        <vater-stack direction="row" justify="center">
+          <watt-spinner />
+        </vater-stack>
+      } @else {
```
Comment (mimse) [inline libs/dh/market-participant/actors/feature-delegation/src/lib/dh-delegation-tab.component.ts:51]: Suggestion: Could else if not be used here?

File: libs/dh/market-participant/actors/feature-delegation/src/lib/dh-delegations.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+export type DhDelegations = DhDelegation[];
+
+export type DhDelegationsByType = {
+  type?: DelegationMessageType;
+  delegations?: DhDelegations;
+};
+
+export type DhDelegationsByDirection = Record<'outgoing' | 'incoming', DhDelegations>;
+
+export type DhDelegationsGrouped = Record<'outgoing' | 'incoming', DhDelegationsByType[]>;
```
Comment (mimse) [inline libs/dh/market-participant/actors/feature-delegation/src/lib/dh-delegations.ts:37]: Praise: nice type handling!

File: libs/dh/market-participant/actors/feature-delegation/src/lib/util/dh-group-delegations.spec.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { DelegationMessageType } from '@energinet-datahub/dh/shared/domain/graphql';
+
+import { DhDelegations } from '../dh-delegations';
+import { dhGroupDelegations } from './dh-group-delegations';
+
+describe(dhGroupDelegations.name, () => {
```
Comment (mimse) [inline libs/dh/market-participant/actors/feature-delegation/src/lib/util/dh-group-delegations.spec.ts:22]: Suggestion: No need for .name

File: libs/dh/market-participant/actors/feature-delegation/src/lib/util/dh-group-delegations.spec.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  it('should group delegations by direction and type', () => {
```
Comment (mimse) [inline libs/dh/market-participant/actors/feature-delegation/src/lib/util/dh-group-delegations.spec.ts:23]: Praise: Nice you are adding tests

File: libs/dh/market-participant/actors/feature-delegation/src/lib/util/dh-group-delegations.ts [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+const outgoingTypes = [DelegationMessageType.Rsm016Outbound, DelegationMessageType.Rsm017Outbound];
+const incomingTypes = [
+  DelegationMessageType.Rsm012Inbound,
+  DelegationMessageType.Rsm014Inbound,
+  DelegationMessageType.Rsm016Inbound,
+  DelegationMessageType.Rsm017Inbound,
+  DelegationMessageType.Rsm019Inbound,
+];
+
+export function dhGroupDelegations(delegations: DhDelegations): DhDelegationsGrouped {
```
Comment (mimse) [inline libs/dh/market-participant/actors/feature-delegation/src/lib/util/dh-group-delegations.ts:35]: Question: Why the dh prefix?

**Review by mimse — CHANGES_REQUESTED** [review]

File: libs/dh/market-participant/actors/feature-delegation/src/lib/dh-delegation-tab.component.ts (hunk 2 above)
Comment (dzhavat) [reply libs/dh/market-participant/actors/feature-delegation/src/lib/dh-delegation-tab.component.ts:51]: Probably can. Nice catch :)

File: libs/dh/market-participant/actors/feature-delegation/src/lib/util/dh-group-delegations.spec.ts (hunk 5 above)
Comment (dzhavat) [reply libs/dh/market-participant/actors/feature-delegation/src/lib/util/dh-group-delegations.spec.ts:23]: Copilot helpe with auto generating them. This makes util functions are quite easy to test.

File: libs/dh/market-participant/actors/feature-delegation/src/lib/util/dh-group-delegations.ts (hunk 6 above)
Comment (dzhavat) [reply libs/dh/market-participant/actors/feature-delegation/src/lib/util/dh-group-delegations.ts:35]: Maybe an just the habit of adding "dh-" prefix to file names, classes, types, etc to things that are public for others to use :)

**Review by mimse — APPROVED** [review]

**Pull request merged**